src/main/java/webserver/controllers/TodoController.java

```diff
@@ -98,10 +99,70 @@ public TodoController(TodoService todoService) {
             .build();
     };
 
+    public Callback<Request, String> getTodoDetailEdit = (request) -> {
+        final int id = getTodoIdFromPath(request.getPath());
+        String html = "";
+        int responseCode = 0;
+
+        Mustache template = MustacheUtil.getTemplate("todo-item-edit.mustache");
+        List<Todo> allTodos = todoService.getTodos();
+        Todo todo = TodosHelper.getTodo(id, allTodos);
+
+        if (todo == null) {
```

Comment:
Here's a good place to use

```java
Optional.ofNullable(todo)
  .map(todo -> MustacheUtil.executeTemplate(template, todo))
  .orElseGet(() -> getTodoNotFound.apply(request));
```

Reply:
^^ and that's where returning `Optional` from `TodoService` can come in handy, since you're probably going to do some sort of null/empty check anyway, after you call it.